### PR TITLE
Improve documentation for Thread

### PR DESCRIPTION
**Pull request by zakstam**
An attempt to improve the documentation for **Thread**
- Adds documentation on how and when the `wait_to_finish` should be used
- Adds a note on what to be careful about when using `wait_to_finish`

I spent **AN HOUR** trying to figure how to reset my local master on the upstream master (i wouldn't accept giving up again. Its not that complicated, it was just my fear of messing up with your branches that kept me back)! Hopefully everything is good (@akien-mga i tried being as descriptive as possible with my commit! 👼).

You can also check the conversation at #46414 for more info.

### CODE REVIEW

File: doc/classes/Thread.xml [hunk 1]
```diff
@@ -48,6 +48,8 @@
 			</return>
 			<description>
 				Joins the [Thread] and waits for it to finish. Returns what the method called returned.
+				Should either be used when you want to retrieve the value returned from the method called by the [Thread] or before freeing the instance that contains the [Thread].
+				NOTE: After the [Thread] finishes joining it will be disposed. If you want to use it again you will have to create a new instance of it.
```
Comment (YeldhamDev) [inline doc/classes/Thread.xml:52]: I would change "NOTE" to "Note" here.

**Review by YeldhamDev — DISMISSED** [review]

File: doc/classes/Thread.xml (hunk 1 above)
Comment (akien-mga) [reply doc/classes/Thread.xml:52]: We usually style it as `[b]Note:[/b] Blabla` (note the colon is also bold).

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
This has been gathering dust for a while. But not anymore! :)

Thanks for this.

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Pull request merged**